File: frontend/src/api/feedback.ts
import { apiClient } from "./client";

// ── Feedback (bug reports / feature requests) ───────────────────────────────

export type FeedbackCategory = "bug" | "feature";

export interface Feedback {
  id: string;
  user_id: string;
  category: FeedbackCategory;
  message: string;
  status: string;
  created_at: string;
}

export async function submitFeedback(
  category: FeedbackCategory,
  message: string,
  includeDiagnostics = true,
): Promise<Feedback> {
  const { data } = await apiClient.post<Feedback>("/feedback", {
    category,
    message,
    include_diagnostics: includeDiagnostics,
  });
  return data;
}

File: backend/models/db/feedback.py
"""Feedback ORM model — user-submitted bug reports and feature requests."""

from __future__ import annotations

import uuid
from datetime import datetime

from sqlalchemy import DateTime, ForeignKey, String, Text, func
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column

from models.db.base import Base


class Feedback(Base):
    __tablename__ = "feedback"

    id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), primary_key=True, default=uuid.uuid4
    )
    user_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), ForeignKey("users.id"), nullable=False, index=True
    )
    # "bug" | "feature"
    category: Mapped[str] = mapped_column(String(20), nullable=False)
    message: Mapped[str] = mapped_column(Text, nullable=False)
    # "open" | "closed"
    status: Mapped[str] = mapped_column(String(20), nullable=False, default="open", index=True)

    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), nullable=False, server_default=func.now(), index=True
    )

    def __repr__(self) -> str:
        return f"<Feedback {self.category} {self.id}>"
